### ISSUE TITLE
Can't open or create nodejs workspace under Ubuntu 16.04

### ISSUE THREAD

**Issue by walkswiththebear** [eranif/codelite#1283]
After upgrading my Ubuntu Linux from 14.04 to 16.04 I cannot open nodejs workspaces anymore. I get a dialog saying "Workspace file is corrupt." When I try to create a new workspace, I don't get the dialog where I can choose a type of workspace. It goes straight to the dialog for choosing a workspace name. The workspace that it creates is not a nodejs workspace. I re-installed codelite, but the problem persists.

**Comment by eranif** [comment]
This means that the WebTools plugin is not loaded.
Can you confirm this (from the plugins menu)

Also, do you get anything in the file ~/.codelite/codelite-stdout-stderr.log ?

**Comment by walkswiththebear** [comment]
When I click on the "Plugins" menu, there are two entries:

Manage Plugins
wxCrafter

When I click on "Manage Plugins", I get a dialog that shows a long list of
plugins, WebTools being one of them. They are all checked, and the "Status"
field has a green circle. But I don't seem to have any of them loaded.

The last three lines of the error log file are:

(codelite:2649): Gtk-CRITICAL **: IA__gtk_widget_set_size_request:
assertion 'width >= -1' failed
sh: 1: cmake: not found

(codelite:2431): Gtk-CRITICAL **: IA__gtk_widget_set_size_request:
assertion 'width >= -1' failed

(codelite:2431): Gtk-CRITICAL **: IA__gtk_widget_set_size_request:
assertion 'width >= -1' failed

The others look very much the same. They're all Gtk errors.

**Comment by eranif** [comment]
This means that WebTools plugin is not loaded (not sure why)
The menu entry should contains long list of plugin
![plugins](https://cloud.githubusercontent.com/assets/3677711/15350919/ffbb0812-1ce4-11e6-8519-815043e28619.png)

Forgot to ask: do you got anything in `~/.codelite/codelite.log` ?

**Comment by walkswiththebear** [comment]
These are the last couple of lines from codelite.log. Again, the others are
very similar. They seem to all be about LLDBDebugger.so.

[ 13:26:58:167 ERR ] Failed to load plugin's dll:
/usr/lib/x86_64-linux-gnu/codelite/LLDBDebugger.so
[ 13:26:58:167 ERR ] liblldb-3.6.so: cannot open shared object file: No
such file or directory
[ 17:47:03:140 ERR ] Failed to load plugin's dll:
/usr/lib/x86_64-linux-gnu/codelite/LLDBDebugger.so
[ 17:47:03:163 ERR ] liblldb-3.6.so: cannot open shared object file: No
such file or directory
[ 22:46:09:506 ERR ] Failed to create single instance socket: CreateServer:
bind operation failed: Address already in use
[ 23:16:23:457 ERR ] Failed to create single instance socket: CreateServer:
bind operation failed: Address already in use

**Comment by eranif** [comment]
I just looked into David's emails (he packages CodeLite for Linux) and he mentioned that Ubuntu 16.04 already comes with CodeLite 9.1
So, I suspect that here what happened:

You installed CodeLite + wxCrafter, however, CodeLite was installed from the official repository while wxCrafter was installed from *our* repository. This is not a problem.

However, there is one slight difference between codelite.org repository and Ubuntu's one: Ubuntu packages splitted the installer into 2: `codelite` and `codelite-plugins` I suspect that you are missing the second package...

try:
`sudo apt-get install codelite-plugins` see if it helps

**Comment by walkswiththebear** [comment]
That fixed it. Thanks!

I have a new problem though, and I'd like to ask you about it although it
is obviously not your problem. When I debug a nodejs application, I cannot
step over any line that is a function call. Say my code looks like this:

foo_1();
foo_2();
foo_3();

I can set a breakpoint on the first line, and the program will stop. But
when I say 'Step over', the output console displays this:

/bin/bash line 1 : 15175 Segmentation fault /usr/bin/nodejs
--debug-brk=5858 <my js filename here>

I just installed a trial version of WebStorm, and the behavior is the same.
So it's obviously not codelite's problem. But still, perhaps you know
something about this? Any help would be much appreciated.

**Comment by eranif** [comment]
I would try a different version of nodejs, maybe its a known issue that has already been fixed?

**Comment by walkswiththebear** [comment]
I have nodejs 4.2.6, and when I run apt update/upgrade, it tells me that
this is the latest one. The problem started when I upgraded from Ubuntu
14.04 to 16.04. Looks like that was a mistake.

**Comment by walkswiththebear** [comment]
Looks like I solved it. I downloaded nodejs 4.4.4 LTS from the nodejs
website and installed it. (It's not a package, it's a tar.xz that you just
unpack into /share or /opt or some such directory.) Now the debugger works
again. Sorry to have bothered you, but perhaps the information comes in
handy, in case somebody else asks the same question.

Thanks again for your quick responses! Much appreciated.
